primitives/src/validator.rs

```diff
@@ -17,14 +19,60 @@ pub enum ValidatorError {
 
 pub type ValidatorFuture<T> = Pin<Box<dyn Future<Output = Result<T, ValidatorError>> + Send>>;
 
-pub trait Validator {
-    fn tick(&self, channel: Channel) -> ValidatorFuture<()>;
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct ValidatorId(#[serde(with = "SerHex::<StrictPfx>")] [u8; 20]);
+
+impl ValidatorId {
+    pub fn inner(&self) -> &[u8; 20] {
+        &self.0
+    }
+
+    pub fn to_hex_non_prefix_string(&self) -> String {
+        hex::encode(self.0)
+    }
+    pub fn to_hex_checksummed_string(&self) -> String {
+        eth_checksum::checksum(&format!("0x{}", hex::encode(self.0)))
+    }
+}
+
+impl TryFrom<&str> for ValidatorId {
+    type Error = DomainError;
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
```

Comment:
Now I realised that we might have a problem here. What if a shorter value is passed? It won't fill up all the bytes and it will create a `ValidatorId` instead of error.

Reply:
A shorter value can't be passed